### DISCUSSION TITLE
How to troubleshoot OpenDTU instances not receiving data ?

### DISCUSSION THREAD

**Discussion by mathieucarbou** [tbnobody/OpenDTU#535] [Q&A] [answered]
Hello,

I bought 3x [ESP32 NodeMCU boards](https://www.amazon.fr/gp/product/B074RGW2VQ) and 3x [NRF24L01+ radio boards](https://www.amazon.fr/gp/product/B08RDJQXVJ) on Amazon (links provided). 

I broke the usb connector of one ESP32 so I have flashed OpenDTU to the 2 other ones I have. 

After having put back the config and flashing https://github.com/tbnobody/OpenDTU/commits/422d489, the 2 OpenDTU are really having issues receiving inverters data. I had the same issue with the first one also: I could only get signal at one specific place at my home, and it was really random. Sometimes I see an inverter, for about 5min, then the DTU is losing it. Same thing for the 2 new boards flashed.

I've tried different combinations with and without the power stabilising module.
And increased the power to max.
Both powered by a 5V 3A adapter.
DTUs are also 3-6m from the inverters, in the garage, without any thick wall.
I have tried with a longer antenna, more directional, but same issue.

I really don't know how to debug this... Any help would be appreciated.

I also don't know how to assert if the problem comes from the ESP32, radio board, antenna, etc.

Thanks for your help!

![PXL_20230124_083240364 NIGHT](https://user-images.githubusercontent.com/61346/214262613-2ea68edf-b378-4877-9985-c112e22ae5cf.jpg)
![PXL_20230124_083314922 NIGHT](https://user-images.githubusercontent.com/61346/214262616-dfb737c3-4e2c-4b92-8c29-7ec248a68be0.jpg)
![PXL_20230124_083120520 NIGHT](https://user-images.githubusercontent.com/61346/214262601-c2d5b89d-6583-4c49-866b-a35f5921c8dd.jpg)
![PXL_20230124_083128978 NIGHT](https://user-images.githubusercontent.com/61346/214262608-e8e0ba85-bd93-4f39-8e90-e67521abf70f.jpg)
![PXL_20230124_083139295 NIGHT](https://user-images.githubusercontent.com/61346/214262612-6ba49010-c626-4ee7-8f3c-8f40daa467b9.jpg)
![image](https://user-images.githubusercontent.com/61346/214261600-25f08787-df63-4887-bedf-625e50310d9b.png)
![PXL_20230124_083039090 NIGHT](https://user-images.githubusercontent.com/61346/214262586-f40e1e7d-fa1d-478d-ac11-e07240d1d6e3.jpg)
![PXL_20230124_083044752 NIGHT](https://user-images.githubusercontent.com/61346/214262597-9e1a279e-170d-4084-abd1-847e2e27e51c.jpg)

**Comment by smarthausleben** [comment, marked as answer]
I also had some trouble to get data from the inverters at the beginning. I could get it fixed by soldering an electrolytic capacitor directly over the power supply pins on the NRF24L01+.

I don’t remember in detail but I guess it worked on gain setting **Minimum (-18db)** adding a 10 µF capacitor. To get it work on **Low (-12db)** I had to add an other 100µF capacitor. The higher gain modes **(-6 dBm)** and **(0 dBm)** did never work. So power supply for the NRF2401+ was the key factor on my setup.

**Reply by battifreund** [reply]
I can confirm that. Came to the same solution. The power supply via the ESP boards is usually not particularly clean and the NRF24L01+ seems to be a diva in this regard.

**Reply by mathieucarbou** [reply]
Thanks! I have those capacitors: several 10 uF and 2x 100 uF. 
If I understand I could then add in parallel 1x 100 uF, or a combination of 100 uF + 10 uF ? Is it worth trying with 2x 100 uF ?
Thanks!

**Reply by mathieucarbou** [reply]
Adding a 100 uF + changing the adapter and position helped a little, but not by much. It is still random. Sometimes I pick the inverters, sometimes not. I really don't know what to try next. These are supposed to work ~50m inside, I am less than 10m away from the farther away inverter.

I'm wondering if the 3 NRF24L01+ modules I got are really what they are supposed to be...

**Reply by smarthausleben** [reply]
I have 10µF an 100µF in parallel. Maybe the polling interval also should not by super short to give a weak power supply a chance to load the capacitor.

**Reply by smarthausleben** [reply]
> Adding a 100 uF + changing the adapter and position helped a little, but not by much. It is still random. Sometimes I pick the inverters, sometimes not. I really don't know what to try next. These are supposed to work ~50m inside, I am less than 10m away from the farther away inverter.
> 
> I'm wondering if the 3 NRF24L01+ modules I got are really what they are supposed to be...

Just try to get it work in Minimum(-18 dBm) first. Having a small and big capacitor in parallel may improve the situation as well.

As I said, I just got it working with -18dBm and -12dBm using a PoE powered ESP32 PoE.

**Reply by mathieucarbou** [reply]
Thanks! I will try next if I see some issues. I was able to get things quite stable now since a few minutes with a dedicated 5V 3A adapter for Raspberry, a 100 uF as close as possible to the radio module and the DTU at a right place im the room, using 5 sec polling and maximum for PA level.
If I see other issues I will reduce the PA level and add a 10uF.
Thanks a lot for your help 👍

**Reply by smarthausleben** [reply]
> Adding a 100 uF + changing the ...

We are dealing with high frequency parts too and if you have a 0.1 µF **ceramic** on hand, try this one in parallel to the 10 µF ceramic.

**Reply by mathieucarbou** [reply]
wait... you meant: 0.1 uF ceramic + 10 uF electrolytic + 100 uF electrolytic ? Or only 0.1 uF ceramic + 10 uF electrolytic ? Turns out I have plenty of 0.1 uF ceramic... I'm glad to have bought this Elegoo starter kit once lol!

**Reply by smarthausleben** [reply]
Just add an 0.1 µF **ceramic** for the higher frequency part.
I would first try it only with 10 µF electrolyte in parallel. Maybe this is enough. If not, then add the 100 µF electrolyte. That’s how I would try to improve it.
